src/lib/pubkey/kyber/kyber_common/kyber.h

```diff
@@ -56,24 +58,16 @@ class BOTAN_PUBLIC_API(3, 0) KyberMode
       Mode m_mode;
    };
 
-enum class KyberKeyEncoding
-   {
-   Raw, // as implemented in the reference implementation
-   Full // as described in draft-uni-qsckeys-00 Section 4.3 (private key), 4.5 (public key)
-   };
-
 class Kyber_PublicKeyInternal;
 class Kyber_PrivateKeyInternal;
 
 class BOTAN_PUBLIC_API(3, 0) Kyber_PublicKey : public virtual Public_Key
    {
    public:
-      Kyber_PublicKey(const std::vector<uint8_t>& pub_key,
-                      KyberMode mode,
-                      KyberKeyEncoding encoding);
+      Kyber_PublicKey(std::span<const uint8_t> pub_key, KyberMode mode);
 
       Kyber_PublicKey(const AlgorithmIdentifier& alg_id,
-                      const std::vector<uint8_t>& key_bits);
+                      std::span<const uint8_t> key_bits);
```

Comment:
It would be nice to make this change for all of the key types before 3.0

(This is the sort of minor thing that can be done after the RC)

Left a comment in #3318